oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -84,7 +84,7 @@ public void setIdToken(String idToken) {
 
   @Override
   public LowLevelHttpRequest buildRequest(String method, String url) throws IOException {
-    if (url.equals(ComputeEngineCredentials.getTokenServerEncodedUrl())) {
+    if (url.startsWith(ComputeEngineCredentials.getTokenServerEncodedUrl())) {
```

Comment:
qq, why does this need to be changed? Is this because for `getRequestMetadata_newAccessTokenWhenScoped`, the first request doesn't have any scopes, but the second request (once `getScoped()`) is called, the request has the query param scopes added to the request?

Reply:
Yes. Basically without this change, mock transport only deal cases with no scopes specified. It will fall back to default behavior and return success code with empty content for the second request with scopes, and eventually hit [this exception](https://github.com/googleapis/google-auth-library-java/blob/cc957e3738dc2f73e86bbef0668f611eb46fe99f/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L287-L289). This change and the added logic below let the mock transport return a different access token when scopes provided.